cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -48,157 +45,113 @@ var (
 	errMissingElasticSearchConfig = errors.New("ElasticSearch not configured")
 )
 
-// SpanHandlerBuilder builds span (Jaeger and zipkin) handlers
-type SpanHandlerBuilder interface {
-	BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error)
+// SpanHandlerBuilder holds configuration required for handlers
+type SpanHandlerBuilder struct {
+	logger         *zap.Logger
+	metricsFactory metrics.Factory
+	collectorOpts  *CollectorOptions
+	storageBuilder storageBuilder
```

Comment:
I may be missing something, but why do you need a function here instead of simply `spanstore.Writer`?